zdm/zdm.py

```diff
@@ -11,13 +11,32 @@ class grid:
 	It also assumes a linear uniform grid.
 	"""
 	
-	def __init__(self):
+	def __init__(self,source_evolution=0,alpha_method=1,luminosity_function=0):
+		"""
+		Class constructor.
+		Source evolution is the function that determines z-dependence.
```

Comment:
Can you also explain what `alpha_method` is?

Reply:
From what I understood from the code, I think this is basically whether we are using the spectral index interpretation of alpha(which is 0) or the rate interpretation(which is 1)

Reply:
Esan is correct. alpha_method=0 is assuming FRBs have a spectrum as F(nu)~nu**-alpha (NOTE: the paper has nu**alpha, but the code still has the original convention!), with Emax essentially fixed at some nu0; in this case, one uses a k-correction for FRBs at redshift z. alpha_method=1 is the 'rate interpretation', where you assumed the spectral index measured by J-P is due to more narrow-band FRBs at low frequency, so while Emax stays constant, as you go to higher redshift, you end up with fewer FRBs. The rate interpretation is quicker to evaluate because when you cycle over alpha you do not have to recalculate your luminosity function.

Reply:
Got it!